app/services/form_response.rb

```diff
@@ -1,21 +1,24 @@
 class FormResponse
-  def initialize(success:, errors: {}, extra: {})
+  attr_reader :errors, :extra, :serialize_error_details_only
+
+  def initialize(success:, errors: {}, extra: {}, serialize_error_details_only: false)
     @success = success
     @errors = errors.is_a?(ActiveModel::Errors) ? errors.messages.to_hash : errors
     @error_details = errors.details if errors.is_a?(ActiveModel::Errors)
     @extra = extra
+    @serialize_error_details_only = serialize_error_details_only
   end
 
-  attr_reader :errors, :extra
-
   def success?
     @success
   end
 
   def to_h
-    { success: success, errors: errors }.merge!(extra).tap do |hash|
-      hash[:error_details] = flatten_details(error_details) if error_details.present?
-    end
+    hash = { success: success }
+    hash[:errors] = errors if !serialize_error_details_only
+    hash[:error_details] = flatten_details(error_details) if error_details.present?
+    hash.merge!(extra)
```

Comment:
I was tempted to change this to make sure that the default values take precedent over `extra`, but that would have been a breaking change from how it was behaving previously.

```suggestion
    hash = { **extra, success: success }
    hash[:errors] = errors if !serialize_error_details_only
    hash[:error_details] = flatten_details(error_details) if error_details.present?
```